src/monaco.contribution.ts

```diff
@@ -121,6 +121,23 @@ const javascriptDefaults = new LanguageServiceDefaultsImpl(
 	{ allowNonTsExtensions: true, allowJs: true, target: ScriptTarget.Latest },
 	{ noSemanticValidation: true, noSyntaxValidation: false });
 
+function getTypeScriptWorker(): monaco.Promise<Worker> {
+	return new monaco.Promise((resolve, reject) => {
+ 		withMode((mode) => {
+ 			mode.getTypeScriptWorker()
+ 				.then(resolve, reject);
+ 		});
+ 	});
+}
+
+function getJavaScriptWorker(): monaco.Promise<Worker> {
+	return new monaco.Promise((resolve, reject) => {
+ 		withMode((mode) => {
+ 			mode.getJavaScriptWorker()
+ 				.then(resolve, reject);
+ 		});
+ 	});

```

Comment:
@paveldk 

i tried following, but result contains the same typescript .. doesn't return the compiled javascript..

 Monaco.languages.typescript.getTypeScriptWorker()
            .then(function(worker) {
                worker(model.uri)
                      .then(function(client) {
                            client.getEmitOutput(model.uri.toString()).then(function(r) {});
                      });
            });